File: Searching algorithms/jump.py
import linear
import math

def jsearch(arr, n):
  step = int(math.sqrt(len(arr)))
  i=step
  while i <=len(arr)-1:
    if arr[i]==n: return i
    elif arr[i]>n: return linear(arr[i-step:i],n)
    elif arr[i]<n: i+=step
  return i

arr = [1,2,3,4,5,6,7,8,9,13,15]
n=int(input())
print(jsearch(arr,n))

File: Searching algorithms/jump.js
function jump(arr,n,l,r){
  function linear(arr,n,l,r){
    for(i=l;i<=r;i++){
      if(arr[i]==n){
        return i;
      }
      else{
        continue;
      }
    }
  }
  function step(arr,n,l,r){
    step = Math.round(Math.sqrt(arr.length-1));
    var i=0;
    while (i<=arr.length-1){
      if (arr[i]==n){
        return i;
      }
      else if (arr[i]<n) {
        i+=step;
      }
      else if (arr[i]>n) {
        return linear(arr,n,i-step,i)
      }
    }
  }
  return step(arr,n,l,r);
}

var arr=[1,5,6,8,12,16,17,26,44,70];
var n=70;
var l = 0;
var r = arr.length-1;
console.log("Your element is: "+ jump(arr,n,l,r));
